tekton/resources/cd/folder-template.yaml

```diff
@@ -0,0 +1,106 @@
+# Copyright 2020 The Tekton Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: tekton.dev/v1alpha1
+kind: TriggerTemplate
+metadata:
+  name: deploy-from-folder
+spec:
+  params:
+  - name: gitRepository
+    description: URL of the repository that holds the folder
+  - name: gitRevision
+    description: Git revision
+  - name: namespace
+    description: Namespace to deploy to in the target cluster
+  - name: clusterResource
+    description: Name of the cluster resource that points to the target cluster
```

Comment:
interesting, i never imagined anyone using a PipelineResource like this - not sure our currently plan with Tasks takes this kinda usage into account ( + @sbwsg )

Reply:
I still like very much the abstraction layer that is given by pipeline resources. 
But there are alternatives for this kind of setup as well.
You could pass around the parameters of the pipeline resource and generate the kubeconfig adding the catalog task at the beginning of the pipeline.
However I feel that the pipeline resource allows to keep the params grouped together and to specify a clearer interface.